packages/stg-evm-v2/ts-src/utils/deploy-usdc.ts

```diff
@@ -57,40 +57,688 @@ const deployUSDC = async (hre: HardhatRuntimeEnvironment, { logger, name, symbol
+            type: 'function',
+        },
+        {
+            inputs: [{ internalType: 'address', name: '_account', type: 'address' }],
+            name: 'unBlacklist',
+            outputs: [],
+            stateMutability: 'nonpayable',
+            type: 'function',
+        },
+        { inputs: [], name: 'unpause', outputs: [], stateMutability: 'nonpayable', type: 'function' },
+        {
+            inputs: [{ internalType: 'address', name: '_newBlacklister', type: 'address' }],
+            name: 'updateBlacklister',
+            outputs: [],
+            stateMutability: 'nonpayable',
+            type: 'function',
+        },
+        {
+            inputs: [{ internalType: 'address', name: '_newMasterMinter', type: 'address' }],
+            name: 'updateMasterMinter',
+            outputs: [],
+            stateMutability: 'nonpayable',
+            type: 'function',
+        },
+        {
+            inputs: [{ internalType: 'address', name: '_newPauser', type: 'address' }],
+            name: 'updatePauser',
+            outputs: [],
+            stateMutability: 'nonpayable',
+            type: 'function',
+        },
+        {
+            inputs: [{ internalType: 'address', name: 'newRescuer', type: 'address' }],
+            name: 'updateRescuer',
+            outputs: [],
+            stateMutability: 'nonpayable',
+            type: 'function',
+        },
+        {
+            inputs: [],
+            name: 'version',
+            outputs: [{ internalType: 'string', name: '', type: 'string' }],
+            stateMutability: 'pure',
+            type: 'function',
+        },
+    ]
     const addressOne = '0x0000000000000000000000000000000000000001'
 
-    const signLibDeploymentName = getUSDCSignatureLibDeployName()
     const implDeploymentName = getUSDCImplDeployName()
     const proxyDeploymentName = getUSDCProxyDeployName()
 
     logger.info(`Deploying USDC token ${symbol} (name ${name})`)
 
-    logger.info(`Deploying USDC SignatureChecker library contract ${signLibContractName} as ${signLibDeploymentName}`)
-    const signatureCheckerLibDeployment = await deploy(signLibDeploymentName, {
-        contract: signLibContractName,
-        from: deployer,
-        args: [],
-        log: true,
-        waitConfirmations: 1,
-        skipIfAlreadyDeployed: true,
-        gasPrice: await hre.ethers.provider.getGasPrice(),
-    })
-
-    logger.info(`Deploying USDC implementation contract ${implContractName} as ${implDeploymentName}`)
-    // Deploy implementation contract
+    // Deploy implementation contract with bytecode
+    logger.info(`Deploying USDC implementation contract as ${implDeploymentName}`)
     const implTokenDeployment = await deploy(implDeploymentName, {
-        contract: implContractName,
+        contract: {
+            // TODO normally expects contract name, but I want to provide bytecode
+            abi: implAbi,
+            bytecode: implBytecode,
+        },
```

Comment:
Please see todo -- I get an error on this line because it expects the contract name string. Is there another way to deploy just the bytecode directly? I will investigate in the meantime!

Reply:
Figured this out\, must use contract factory